cmp/grouping/GroupingChooserModel.ts

```diff
@@ -84,8 +84,9 @@ export class GroupingChooserModel extends HoistModel {
 
     @observable.ref favorites: string[][] = [];
 
-    dimensions: Record<string, DimensionSpec>;
-    dimensionNames: string[];
+    @observable.ref private _dimensions: Record<string, DimensionSpec>;
+    @observable.ref private dimensionNames: string[];
```

Comment:
Happy to discuss making these private - I asked Ryan to rename/make private `_dimensions` so it was clear that you must call `setDimensions` with its validation / pruning logic.

I then made `dimensionNames` private - did not seem likely that it needed to be public, although if we are concerned apps might be relying on it for some reason...  Wanted to keep some symmetry here.

Not sure how to best to tighten down privacy settings like this and harden classes - is an incremental approach helpful, or does it start to add more confusion than it is worth, without taking on a full review of each property and method ?